### DISCUSSION TITLE
Local Developement Sucks

### DISCUSSION THREAD

**Discussion by Akash187** [supabase/supabase#18975] [General]
I have reported an [Issue with logging in to Local Development](https://github.com/supabase/supabase/issues/18707) It's been more than a week and nobody has time even to take a look.

Logging is an essential feature for development, and its current state is hindering my progress. Without proper logs, it's challenging to identify where the code is failing.

Until now, using Supabase remotely has provided a fast and smooth development experience. However, transitioning to Local Development has been quite challenging. It seems that the local development environment is still in its early stages and needs further refinement. I am spending a considerable amount of time configuring tools and attempting to work without the support of logging, which is causing significant delays.

Thank you

**Comment by ffd114** [comment]
A bug on local development also affected self hosted deployment, see #18668, it is very annoying because we have to rely on others means of logging. I agree that they need to step up their local development experience, especially if it affected the self hosted. Right now I use their managed services, but I'm preparing for self hosting one day, in case a project require me to host on specific instance.

Just some rant from me who likes this project very much but feels annoyed with these bugs.

**Comment by Akash187** [comment]
@ffd114 I think they are slowing development so that they can sell their cloud service. I have no issue but when you are selling Supabase as open source and available for self host then you should deliver that too.

**Comment by Ziinc** [comment]
RE the local logging, if there are blockers you can use the logs display within the docker desktop as an alternative display while bugs are getting fixed.

Local dev is very much an important part of the experience and stack, but given how fast the product moves and launches new features, some teething bugs are bound to be found despite the teams best efforts. If there are feature requests or ux improvement suggestions, such constructive feedback is most appreciated.

moving to discussion.

**Comment by raman04-byte** [comment]
Any update on this issue. Is it got fixed?
